// @faire/mjml-react/extensions/mjml-tracking-pixel.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.MjmlTrackingPixel = void 0;
const react_1 = __importStar(require("react"));
const __1 = require("..");
class MjmlTrackingPixel extends react_1.Component {
    render() {
        const { src } = this.props;
        const trackingPixelStyle = {
            display: "table",
            height: "1px!important",
            width: "1px!important",
            border: "0!important",
            margin: "0!important",
            padding: "0!important",
        };
        return (react_1.default.createElement(__1.MjmlRaw, null,
            react_1.default.createElement("img", { src: src, style: trackingPixelStyle, width: 1, height: 1 })));
    }
}
exports.MjmlTrackingPixel = MjmlTrackingPixel;
