// @faire/mjml-react/utils/render.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.render = void 0;
const html_minifier_1 = require("html-minifier");
const mjml_1 = __importDefault(require("mjml"));
const renderToMjml_1 = require("./renderToMjml");
function render(email, options = {}) {
    const defaults = {
        keepComments: false,
        beautify: false,
        validationLevel: "strict",
    };
    const parseResults = (0, mjml_1.default)((0, renderToMjml_1.renderToMjml)(email), Object.assign(Object.assign(Object.assign({}, defaults), options), { minify: undefined }));
    if (options.minify) {
        return {
            html: (0, html_minifier_1.minify)(parseResults.html, {
                caseSensitive: true,
                collapseWhitespace: true,
                minifyCSS: true,
                removeComments: true,
                removeEmptyAttributes: true,
            }),
        };
    }
    return parseResults;
}
exports.render = render;
